src/app/embed/adapters/feed.adapter.ts

```diff
@@ -0,0 +1,88 @@
+import {Injectable} from '@angular/core';
+import { Http, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import { AdapterProperties, DataAdapter } from './adapter.properties';
+import { sha1 }  from './sha1'
+
+const GUID_PREFIX = 's1!'
+@Injectable()
+export class FeedAdapter implements DataAdapter {
+	private feedUrl: string
+	private feedId: string
+	private guid: string
+
+  constructor(private http: Http) {}
+
+	public getProperties(params): Observable<AdapterProperties> {
+    this.feedUrl = params[EMBED_FEED_URL_PARAM]
+    this.feedId  = params[EMBED_FEED_ID_PARAM]
+    this.guid    = params[EMBED_EPISODE_GUID_PARAM]
+
+ 		return this.http.get(this.draperUrl).map((res: Response) => {
+      let xml = res.text();
+      let parser = new DOMParser();
+      let doc = <XMLDocument> parser.parseFromString(xml, 'application/xml');
+      let rpNamespace = doc.lookupNamespaceURI('rp')
+      let atomNamespace = doc.lookupNamespaceURI('atom')
+      let elements = doc.querySelectorAll('item');
+
+      let episode;
+      if(this.guid) {
+        for (let i = 0; i < elements.length; ++i) {
+          let item = <Element> elements[i];
+          let episodeGuid = item.getElementsByTagName('guid')[0].textContent;
+          if(this.isEncoded(this.guid) && !this.isEncoded(episodeGuid)){
+            episodeGuid = this.encodeGuid(episodeGuid);
+          };
+          if(episodeGuid.indexOf(this.guid) !== -1){
+            episode = item;
+						break;
+          }
+        };
+      }
+			episode = episode || elements[0]
+
+      let title = episode.getElementsByTagName('title')[0].textContent
+      let audioUrl = episode.getElementsByTagName('enclosure')[0].getAttribute('url')
+      let subtitle = doc.getElementsByTagName('title')[0].textContent
+      let subscribeUrl = doc.getElementsByTagNameNS(atomNamespace, 'link')[0].getAttribute('href')
+      // Future feeds may not include RP-namespaced data!
+      let feedArtworkUrl  = doc.getElementsByTagNameNS(rpNamespace, 'image')[0].getAttribute('href')
+      let artworkUrl  = episode.getElementsByTagNameNS(rpNamespace, 'image')
+
+      if (artworkUrl.length == 0){
+        artworkUrl = feedArtworkUrl
+      } else {
+        artworkUrl = artworkUrl[0].getAttribute('href')
+      };
+
+      return {
+        audioUrl,
+        title,
+        subtitle,
+        subscribeUrl,
+        feedArtworkUrl,
+        artworkUrl
+      }
+    }).catch((err, caught) => Observable.of({}));
+	}
+
+  private isEncoded(guid): boolean {
+    return (guid.indexOf(GUID_PREFIX) == 0)
+  }
+
+  private encodeGuid(guid): string {
+    return `${GUID_PREFIX}${sha1.hash(guid)};`
+  }
+
+	private get draperUrl(): string {
+    let feedUrl, feedId;
+    if (this.feedUrl) {
+      feedUrl = `${window['ENV']['FEED_PROXY_URL']}${this.feedUrl}`
+    } else if (this.feedId) {
+      feedId = `https://draper.radiopublic.com/transform?program_id=${this.feedId}`
```

Comment:
I think we'll need to break params out if we go this route, because you don't want to have both a FeedAdapter and a DraperAdapter running simultaneously, and the DraperAdapter still requires the EMBED_FEED_URL_PARAM. The thinking behind putting these in the same adapter is that if there is a draper id available, it skips the EMBED_FEED_URL_PARAM provided url, but otherwise it fetches that (so you aren't fetching a feed in 2 ways simultaneously)

That said, seems like maybe we could refactor the draper adapter to not require the EMBED_FEED_URL_PARAM and instead pull it from the draper feed response.